httpcache.go

```diff
@@ -227,9 +229,22 @@ func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error
 				resp.Header.Set(fakeHeader, reqValue)
 			}
 		}
-		respBytes, err := httputil.DumpResponse(resp, true)
-		if err == nil {
-			t.Cache.Set(cacheKey, respBytes)
+		switch req.Method {
+		case "GET":
+			// Delay caching until EOF is read.
+			resp.Body = newCachedReadCloser(resp.Body, func(r io.ReadCloser) {
+				response := *resp
```

Comment:
Personally, I wouldn't be afraid of doing `resp := *resp`. I think its meaning should be well understood by Go programmers. See last code block of https://golang.org/doc/faq#closures_and_goroutines.